app/src/main/java/com/example/vybe/AddEdit/AddEditController.java

```diff
@@ -0,0 +1,173 @@
+package com.example.vybe.AddEdit;
+
+import android.graphics.Bitmap;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.target.CustomTarget;
+import com.bumptech.glide.request.transition.Transition;
+import com.example.vybe.Models.SocSit;
+import com.example.vybe.Models.Vibe;
+import com.example.vybe.Models.VibeEvent;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.io.ByteArrayOutputStream;
+import java.util.HashMap;
+
+public class AddEditController {
+    private static final String TAG = "AddEditController";
+    private static AddEditController instance;
+
+    private AddEditVibeEventActivity activity;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
+    private StorageReference storageRef = FirebaseStorage.getInstance().getReference();
+    private String vibeEventDBPath = "Users/" + mAuth.getCurrentUser().getUid() + "/VibeEvents";
+    private VibeEvent vibeEvent;
+
+    private AddEditController() {
+    }
```

Comment:
Shouldn't these be intitalized in the constructor?

Reply:
I don't think it really makes a difference either way